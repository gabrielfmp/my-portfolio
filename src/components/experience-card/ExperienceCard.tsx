import {
  DescriptionItem,
  DescriptionList,
  ExperienceCardWrapper,
  MiniDot,
  Role,
  ThinText,
  TimeAndPlaceWrapper,
} from './styles';
import React, { FunctionComponent } from 'react';

const ExperienceCard: FunctionComponent<{
  description: Array<string>;
  endYear?: number;
  noPaddingBottom?: boolean;
  place: string;
  role: string;
  startYear: number;
}> = ({ description, endYear, noPaddingBottom, startYear, place, role }) => (
  <ExperienceCardWrapper noPaddingBottom={noPaddingBottom}>
    <Role>{role}</Role>

    <TimeAndPlaceWrapper>
      <ThinText>{place}</ThinText>
      <MiniDot />
      <ThinText>{`${startYear} - ${endYear || 'Present'}`}</ThinText>
    </TimeAndPlaceWrapper>

    <DescriptionList>
      {description.map((description, i) => (
        <DescriptionItem key={i}>{description}</DescriptionItem>
      ))}
    </DescriptionList>
  </ExperienceCardWrapper>
);

export default ExperienceCard;
