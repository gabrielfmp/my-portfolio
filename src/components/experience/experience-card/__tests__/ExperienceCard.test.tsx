import { render } from '@testing-library/react';
import ExperienceCard from '../ExperienceCard';
import React from 'react';

test('snapshot', () => {
  const { asFragment } = render(
    <ExperienceCard
      role="foo"
      place="bar"
      startYear={2025}
      description={['foo', 'bar']}
    />
  );

  expect(asFragment()).toMatchSnapshot();
});
