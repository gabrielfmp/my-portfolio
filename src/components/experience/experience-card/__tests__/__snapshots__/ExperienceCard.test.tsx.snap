// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot 1`] = `
<DocumentFragment>
  <div
    class="css-sty4r9"
  >
    <p
      class="css-11bj48"
    >
      foo
    </p>
    <div
      class="css-14pnp1n"
    >
      <p
        class="css-zkfum4"
      >
        bar
      </p>
      <div
        class="css-15ag3i8"
      />
      <p
        class="css-zkfum4"
      >
        2025 - Present
      </p>
    </div>
    <ul
      class="css-1rdmnzn"
    >
      <li
        class="css-1g7zcol"
      >
        foo
      </li>
      <li
        class="css-1g7zcol"
      >
        bar
      </li>
    </ul>
  </div>
</DocumentFragment>
`;
